### PR TITLE
Feast Core: Stage files manually when launching Dataflow jobs

### PR DESCRIPTION
**Pull request by davidheryanto**
To resolve error when launching Dataflow jobs from Feast Core:
```
Caused by: java.lang.IllegalArgumentException: Unable to convert url (jar:file:/opt/feast/core/target/feast-core-0.3.0-SNAPSHOT.jar!/BOOT-INF/classes!/) to file.
	at org.apache.beam.runners.core.construction.PipelineResources.detectClassPathResourcesToStage(PipelineResources.java:67)
	at org.apache.beam.runners.dataflow.DataflowRunner.fromOptions(DataflowRunner.java:285)
	... 38 common frames omitted
```
When Feast Core starts from a packaged jar.

Reference:
https://stackoverflow.com/questions/48961440/illegalargumentexception-unable-to-convert-url-jarfile-app-jar-boot-inf-cla

https://stackoverflow.com/questions/48292491/java-dataflow-unable-to-use-classloader-to-detect-classpath-elements

### CODE REVIEW

File: core/src/main/java/feast/core/job/dataflow/DataflowJobManager.java [hunk 1]
```diff
@@ -151,6 +154,19 @@ private ImportOptions getPipelineOptions(String jobName, List<FeatureSetSpec> fe
...
     pipelineOptions.setJobName(jobName);
+
+    // "DataflowRunner.detectClassPathResourcesToStage" runs into this issue when trying to detect jars
+    // to upload, when Feast Core starts from a packaged jar.
+    //
+    // https://stackoverflow.com/questions/48961440/illegalargumentexception-unable-to-convert-url-jarfile-app-jar-boot-inf-cla
+    // https://stackoverflow.com/questions/48292491/java-dataflow-unable-to-use-classloader-to-detect-classpath-elements
+    //
+    // Set the files to stage manually, following the quick workaround in Stackoverflow.
+    // Perhaps there is a cleaner solution?
```
Comment (woop) [inline core/src/main/java/feast/core/job/dataflow/DataflowJobManager.java]: I don't feel like this final question adds any value

**Comment by davidheryanto** [comment]
/hold
Stil have issues when the Dataflow job is starting up.
```
 A JNI error has occurred, please check your installation and try again
Caused by: java.lang.ClassNotFoundException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.graph.Network
```

**Comment by davidheryanto** [comment]
New approach that _unpack the packaged jar and stage the files manually_ when trying to submit Dataflow jobs from Feast Core seems to work. 

But the extract jar method is **too slow** . Takes more than 5 minutes to unpack Feast Core jar using `extractJar` method in this commit. 

In comparison using `unar` [command ](https://github.com/incbee/Unarchiver) takes less than 10 seconds to unpack the jar.

File: core/src/main/java/feast/core/job/dataflow/DataflowJobManager.java (hunk 1 above)
Comment (davidheryanto) [reply core/src/main/java/feast/core/job/dataflow/DataflowJobManager.java]: No longer using this method because it wasn't working.

**Comment by davidheryanto** [comment]
Current workaround for the slow unpacking of jar when running Feast in production:
https://github.com/gojek/feast/pull/289/commits/9ba7c90ddb2d2bb0499e6695ea86ec11432f1181

**Pull request merged**